resources/Microsoft.Windows.Setting.Accessibility/Microsoft.Windows.Setting.Accessibility.psm1

```diff
@@ -584,6 +595,128 @@ class TextCursor
+            return ($colorFilterSetting -eq ([ColorFilter]::FilterStatusValue))
+        }
+    }
+
+    static [int] GetColor()
+    {
+        $FilterColorArgs = @{  Path = $global:ColorFilteringRegistryPath; Name = ([ColorFilter]::FilterColorProperty)}
+        if (-not(DoesRegistryKeyPropertyExist @FilterColorArgs))
+        {
+            return $false
+        }
+        else
+        {
+            $colorFilterSetting = (Get-ItemProperty @FilterColorArgs).FilterType
+            return $colorFilterSetting
+        }        
+    }
+
+    static [nullable[bool]] GetKeyboardShortcutStatus()
+    {
+        $KeyboardShortcutStatusArgs = @{ Path = $global:ColorFilteringRegistryPath; Name = ([ColorFilter]::KeyboardShortcutStatusProperty); }
+        if (-not(DoesRegistryKeyPropertyExist @KeyboardShortcutStatusArgs))
+        {
+            return $false
+        }
+        else
+        {
+            $colorFilterSetting = (Get-ItemProperty @KeyboardShortcutStatusArgs).HotkeyEnabled
+            return $colorFilterSetting
+        }        
+    }
+
+    [ColorFilter] Get()
+    {
+        $currentState = [ColorFilter]::new()
+        $currentState.FilterStatus = [ColorFilter]::GetStatus()
+        $currentState.FilterColor = [ColorFilter]::GetColor()
+        $currentState.KeyboardShortcutStatus = [ColorFilter]::GetKeyboardShortcutStatus()
+        
+        return $currentState
+    }
+
+    [bool] Test()
+    {
+        $currentState = $this.Get()
+        if (($null -ne $this.FilterStatus) -and ($this.FilterStatus -ne $currentState.FilterStatus))
+        {
+            return $false
+        }
+        if ((0 -ne $this.FilterColor) -and ($this.FilterColor -ne $currentState.FilterColor))
+        {
+            return $false
+        }
+        if (($null -ne $this.KeyboardShortcutStatus) -and ($this.KeyboardShortcutStatus -ne $currentState.KeyboardShortcutStatus))
+        {
+            return $false
+        }
+
+        return $true
+    }
+
+    [void] Set()
+    {
+        if (-not $this.Test())
+        {
+            if ($null -ne $this.FilterStatus) 
+            {
+                $FilterStatusArgs = @{ Path = $global:NTAccessibilityRegistryPath; Name = ([ColorFilter]::FilterStatusProperty); }
+                $ColorFilterValue = if ($this.FilterStatus) { ([ColorFilter]::FilterStatusValue) } else { "" }
+                Set-ItemProperty @FilterStatusArgs -Value $ColorFilterValue
+            }
+            
+            if (0 -ne $this.FilterColor) 
+            {
+                $FilterColorArgs = @{  Path = $global:ColorFilteringRegistryPath; Name = ([ColorFilter]::FilterColorProperty)}
+                $min = 1
+                $max = 99999999
+                if ($this.FilterColor  -notin $min..$max) 
+                { 
+                    throw "FilterColor must be between $min and $max. Value $($this.FilterColor) was provided." 
+                }

```

Comment:
$this.FilterColor should be an enum so the value should already be valid. You just need to convert it to the proper DWORD value for the registry. Checking if it is within a range is not necessary

Reply:
You've already mentioned this above. Does mentioning it twice add value? https://github.com/microsoft/winget-dsc/pull/66#discussion_r1802039890